### ISSUE TITLE
Tuist source glob doesn't handle nested folder symlinks

### ISSUE THREAD

**Issue by danyf90** [tuist/tuist#3268]
**Describe the bug**
If a Tuist source glob contains a symlinked folder, it's root files are included in the project, but files included in its subfolder are not included.

The problems seems to be in `Glob.swift:138`, as `fileManager.subpathsOfDirectory` doesn't return the `SymLinked/Nested` folder.
A similar scenario works fine in SPM, so it might create issues in our SPM integration.

**To Reproduce**
1. Download the attached project
2. Run `tuist generate` inside the `Project` folder
3. See that only top-level `value1.swift` file is included, while nested `SymLinked/value2.swift` is not, and hence build fails because some files are not present

**Expected behavior**
Nested files should be imported correctly and build should succeeds.

**Comment by fortmarek** [comment]
Interesting!

Does [this](https://github.com/tuist/tuist/blob/main/Sources/TuistSupport/Utils/Glob.swift#L131) return false when it's a symlinked folder? If so, I think we can simply change it to `guard isDirectory(path: FileHandler.shared.resolveSymlinks(fullPath))...`. I believe it should work then.

**Comment by danyf90** [comment]
Nope, the isDirectory works fine, the problem is that we don't even pass from there as symlinked subdirectories are not returned by `fileManager.subpathsOfDirectory`

**Comment by adellibovi** [comment]
hi @danyf90, I was trying out your example with SPM and Xcode and I dont get to make it resolve the subpaths, maybe I am missing something? Are we sure that in SPM this works? :thinking_face:

**Comment by danyf90** [comment]
Yep, I have a similar scenario in which SPM build fines, I'll try to provide a Package.swift for the attached project which would build fine
